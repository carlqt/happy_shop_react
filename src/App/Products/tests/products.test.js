import React from 'react';
import { shallow, mount } from 'enzyme';

import ProductIndex from '../products';

describe('Products Index', () => {
  const props = {
    getProducts: jest.fn(),
    setFilters: jest.fn(),
    products: [],
    meta: {},
    queryParams: {
      sort: '',
      categories: [],
      priceRange: [0, 30000],
      page: 1,
    },
  };

  it('renders correctly', () => {
    const productIndex = shallow(<ProductIndex {...props} />).dive();
    expect(productIndex).toMatchSnapshot();
  }),
  describe('formatFilterParams', () => {
    const productIndex = shallow(<ProductIndex {...props} />).dive();
    const params = {
      priceRange: [700, 2000],
      sort: "price"
    };

    const formatParams = productIndex.instance().formatFilterParams(params)
    expect(formatParams).toEqual({
      price: '700..2000',
      sort: 'price',
    });
  })
})
