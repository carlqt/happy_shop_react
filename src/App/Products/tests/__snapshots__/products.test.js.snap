// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Index renders correctly 1`] = `
<WithStyles(Grid)
  className="ProductList-container-1"
  container={true}
  spacing={0}
>
  <WithStyles(Grid)
    item={true}
    xs={3}
  >
    <WithStyles(FilterSection)
      loadProducts={[Function]}
      meta={Object {}}
      queryParams={
        Object {
          "categories": Array [],
          "page": 1,
          "priceRange": Array [
            0,
            30000,
          ],
          "sort": "",
        }
      }
      setFilters={[MockFunction]}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    className="ProductList-drawer-2"
    item={true}
    xs={9}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <label>
        Sort By:
        <select
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
             -- no sort -- 
          </option>
          <option
            value="price"
          >
            Price (low to high)
          </option>
          <option
            value="-price"
          >
            Price (high to low)
          </option>
        </select>
      </label>
      <WithStyles(Pagination)
        meta={Object {}}
        pageUpdate={[Function]}
        queryParams={
          Object {
            "categories": Array [],
            "page": 1,
            "priceRange": Array [
              0,
              30000,
            ],
            "sort": "",
          }
        }
        setFilters={[MockFunction]}
      />
      <WithStyles(Typography)
        style={
          Object {
            "display": "inline-block",
          }
        }
        variant="h6"
      />
    </WithStyles(Grid)>
    <WithStyles(List)
      data={Array []}
      onItemClick={[Function]}
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
